internal/server/routes.go

```diff
@@ -108,8 +109,18 @@ func (s *Server) GenerateRoutes(promRegistry prometheus.Registerer) Routes {
 
 	// Deprecated in 0.12
 	// TODO: remove after a couple versions
-	get(a, authn, "/v1/users/:id/grants", a.ListUserGrants)
-	get(a, authn, "/v1/groups/:id/grants", a.ListGroupGrants)
+	add(a, authn, route[api.Resource, *api.ListResponse[api.Grant]]{
+		method:       http.MethodGet,
+		path:         "/v1/users/:id/grants",
+		handler:      a.ListUserGrants,
+		omitFromDocs: true,
+	})
+	add(a, authn, route[api.Resource, *api.ListResponse[api.Grant]]{
+		method:       http.MethodGet,
+		path:         "/v1/groups/:id/grants",
+		handler:      a.ListGroupGrants,
+		omitFromDocs: true,
+	})
```

Comment:
Is your plan that we update all routes to the `add` func rather than `get, post, ...`?

Reply:
Good question! I was not planning on doing that. The reason these are special cases is because they use `omitFromDocs: true`. We could hide some of this with a `deprecated` helper (similar to `get`, `post`, etc) that accepts the method, path, and handler.

I think the `get`, `post`, etc helpers are nice and keep the route definitions concise and readable.

One thing I might look at would be some way to include the api migrations / previous versions along with the new canonical route, so that we have separate lists per method+path, instead of a single list for everything. I'm not sure what that would look like yet, but hopefully those could be passed along to the `get`, `post`, etc functions.